lib/circle_flags.dart

```diff
@@ -1,76 +1,78 @@
 library circle_flags;
 
 import 'package:flutter/material.dart';
-import 'package:flutter/services.dart';
-import 'package:flutter_svg/flutter_svg.dart';
-import 'package:flutter_svg/parser.dart';
+import 'package:jovial_svg/jovial_svg.dart';
 
 /// a rounded flag
 class CircleFlag extends StatelessWidget {
   final String countryCode;
-  final double? size;
-  final SvgParser svgParser = SvgParser();
-  late final String assetName;
+  final double size;
 
-  CircleFlag(this.countryCode, {Key? key, this.size}) : super(key: key) {
-    assetName =
-    'packages/circle_flags/assets/svg/${countryCode.toLowerCase()}.svg';
-  }
+  final Widget Function(BuildContext)? onLoading;
+
+  final Color? placeholderBackgroundColor;
+  final Color? placeholderTextColor;
+
+  const CircleFlag(this.countryCode,
+      {Key? key,
+      this.size = 48,
+      this.onLoading,
+      this.placeholderBackgroundColor,
+      this.placeholderTextColor})
+      : super(key: key);
 
   @override
   Widget build(BuildContext context) {
-    final size = this.size ?? 48;
-
     return SizedBox(
       width: size,
       height: size,
-      child: FutureBuilder(
-        future: _isAssetNameValidSvg(),
-        builder: (context, AsyncSnapshot<bool> snapshot) {
-          if (snapshot.hasData && snapshot.data == true) {
-            return SvgPicture.asset(
-              assetName,
-              width: size,
-              height: size,
-            );
-          }
-
-          if (snapshot.hasError) {
-            return Container(
-              width: size,
-              height: size,
-              padding: const EdgeInsets.all(8),
-              decoration: const BoxDecoration(
-                shape: BoxShape.circle,
-                color: Colors.grey,
-              ),
-              child: Text(
-                countryCode,
-                textAlign: TextAlign.center,
-                style: const TextStyle(
-                  color: Colors.white,
-                ),
-              ),
-            );
-          }
-
-          return SizedBox(
-            width: size,
-            height: size,
-            child: const CircularProgressIndicator(),
-          );
-        },
+      child: ScalableImageWidget.fromSISource(
+        si: ScalableImageSource.fromSI(
+            DefaultAssetBundle.of(context),
+            'packages/circle_flags/assets/jovial/${countryCode.toLowerCase()}.si'
+        ),
+        onLoading: onLoading ?? _buildPlaceholder,
```

Comment:
In my opinion the place holder should be an empty container by default. It's really not that long until it shows up and having something blinking is more annoying